### PR TITLE
Improve pre-release version deployment

### PR DESCRIPTION
**Pull request by rubo**
## Changes

- Updated release workflow to handle pre-release versions
- Publishing pre-release versions to GitHub and Docker Hub only
- Require approval for any step after the build: `Build -> Approval -> Publish to Homebrew, GitHub, Downloads, Docker, PPA`
- Revised some namings

#### What types of changes does your code introduce?

- [ ] Bugfix (a non-breaking change that fixes an issue)
- [ ] New feature (a non-breaking change that adds functionality)
- [ ] Breaking change (a fix or a feature that would cause existing functionality not to work as expected)
- [ ] Documentation update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional or API changes)
- [x] Build-related changes
- [ ] Other: _description_ 

#### Requires testing

- [x] Yes
- [ ] No

#### If yes, did you write tests?

- [ ] Yes
- [x] No

#### Notes on testing

To test, tag a pre-release like `x.y.z-label` and run the workflow.

## Remarks

Maybe it's worth adding a default tag for pre-release Docker images. Say, `latest-preview`, `latest-unstable`, or just `unstable`.

### CODE REVIEW

**Review by kamilchodola — APPROVED** [review]

File: .github/workflows/release-nethermind.yml [hunk 1]
```diff
@@ -205,12 +210,12 @@ jobs:
...
+      if: needs.build.outputs.PRERELEASE == 'false'
       run: |
         docker buildx build --platform=linux/amd64,linux/arm64 -t "${{ env.DOCKER_IMAGE }}:latest" -t "${{ env.DOCKER_IMAGE }}:${{ github.event.inputs.tag }}" -f Dockerfile --build-arg COMMIT_HASH=${{ needs.build.outputs.COMMIT_HASH }} --build-arg BUILD_TIMESTAMP=${{ needs.build.outputs.BUILD_TIMESTAMP }} . --push
         curl -s -X POST -u "${{ secrets.REPOSITORY_DISPATCH_TOKEN }}" -H "Accept: application/vnd.github.everest-preview+json" -H "Content-Type: application/json" -d '{"event_type":"dappnode","client_payload":{"tag":"${{ github.event.inputs.tag }}"}}' https://api.github.com/repos/$GITHUB_REPOSITORY/dispatches
     - name: Build and push image to Docker registry  (patch)
-      if: ${{ contains(github.event.inputs.tag, '-') }}
+      if: needs.build.outputs.PRERELEASE == 'true'
       run: |
         docker buildx build --platform=linux/amd64,linux/arm64 -t "${{ env.DOCKER_IMAGE }}:${{ github.event.inputs.tag }}" -f Dockerfile --build-arg COMMIT_HASH=${{ needs.build.outputs.COMMIT_HASH }} --build-arg BUILD_TIMESTAMP=${{ needs.build.outputs.BUILD_TIMESTAMP }} . --push
```
Comment (FalcoXYZ) [inline .github/workflows/release-nethermind.yml:220]: Do we want to deploy this to an unstable tag too? CC @kamilchodola

File: .github/workflows/release-nethermind.yml [hunk 2]
```diff
@@ -85,11 +86,23 @@ jobs:
           ${{ github.workspace }}/${{ env.PACKAGE_DIR }}/
           ${{ env.SCRIPTS_PATH }}/
         key: packages-${{ steps.build-runner.outputs.COMMIT_HASH }}
+        
+  approval:
```
Comment (FalcoXYZ) [inline .github/workflows/release-nethermind.yml:90]: How does this approval work?

Comment (rubo) [reply .github/workflows/release-nethermind.yml:90]: This is `trigger-publish` renamed to `approval`.

File: .github/workflows/release-nethermind.yml (hunk 1 above)
Comment (rubo) [reply .github/workflows/release-nethermind.yml:220]: I also think we may need it. If not now, maybe later.

Comment (FalcoXYZ) [reply .github/workflows/release-nethermind.yml:220]: Add it, I don't see a reason to not deploy it to 'unstable' tag as well. So we will have two tags for RC: 

'unstable' and example: '1.15.0-RC1' (or however you're naming the version)

File: .github/workflows/release-nethermind.yml (hunk 2 above)
Comment (rubo) [reply .github/workflows/release-nethermind.yml:90]: Everything but Homebrew. Now it does as well.

Comment (FalcoXYZ) [reply .github/workflows/release-nethermind.yml:90]: OK, it's good then. Can mark it as resolved.

File: .github/workflows/release-nethermind.yml (hunk 1 above)
Comment (rubo) [reply .github/workflows/release-nethermind.yml]: Added

**Review by FalcoXYZ — APPROVED** [review]
LGTM

**Pull request merged**